WarehouseManagement.Api/Controllers/ZoneController.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.Security.Claims;
+using Microsoft.AspNetCore.Mvc;
+using WarehouseManagement.Core.Contracts;
+using WarehouseManagement.Core.DTOs.Zone;
+
+namespace WarehouseManagement.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ZoneController : ControllerBase
+    {
+        private readonly IZoneService zoneService;
+
+        public ZoneController(IZoneService zoneService)
+        {
+            this.zoneService = zoneService;
+        }
+
+        [HttpGet("{id}")]
+        [ProducesResponseType(200, Type = typeof(ZoneDto))]
+        [ProducesResponseType(404)]
+        public async Task<IActionResult> GetById(int id)
+        {
+            try
+            {
+                ZoneDto model = await this.zoneService.GetByIdAsync(id);
+
+                return Ok(model);
+            }
+            catch (KeyNotFoundException e)
+            {
+                return NotFound(e.Message);
+            }
```

Comment:
the others havent used `try/catch` inside the controllers but rather check if `GetByIdAsync` returned something otherwise return `NotFound` so maybe to keep consistency look into how they have already started and change the controller and service methods.